provider/oci/instance.go

```diff
@@ -265,7 +266,7 @@ func (o *ociInstance) waitForMachineStatus(state ociCore.InstanceLifecycleStateE
 				"Timed out waiting for instance to transition from %v to %v",
 				o.raw.LifecycleState, state,
 			)
-		case <-o.env.clock.After(10 * time.Second):
+		case <-o.env.clock.After(pollTime * time.Second):
```

Comment:
I don't think this does what you want. I think you want to declare the timeout outside of the for{} loop, otherwise the timer is reset every time through. Is that what you want?

Reply:
This is the polling timeout. Inside the for loop you have a select with 2 conditions. One for the [timeout timer](https://github.com/juju/juju/pull/8540/files#diff-b21db58f26cb98e25d5111e6d5f98438R259), which is outside of the loop and one that simply does a "sleep", before polling the status.

Reply:
ah gotcha. ok